gnovm/stdlibs/std/coins.gno

```diff
@@ -13,21 +15,77 @@ type Coin struct {
 	Amount int64  `json:"amount"`
 }
 
+// NewCoin returns a new coin with a denomination and amount
+func NewCoin(denom string, amount int64) Coin {
+	return Coin{
+		Denom:  denom,
+		Amount: amount,
+	}
+}
+
+// String provides a human-readable representation of a coin
 func (c Coin) String() string {
+	if c.IsZero() {
+		return ""
+	}
+
 	return strconv.Itoa(int(c.Amount)) + c.Denom
 }
 
+// IsGTE returns true if they are the same type and the receiver is
+// an equal or greater value
 func (c Coin) IsGTE(other Coin) bool {
 	if c.Denom != other.Denom {
-		panic("invalid coin denominations: " + c.Denom)
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
 	}
```

Comment:
returning errors here would be great for easier unit testing.

Reply:
Fair point, however:

This was copied over from the tm2 version of coins. The issue with this is that we do not have support for Errorf or fmt in stdlibs, so the dynamic parsing of the errors would be a pain. WDYT?

Reply:
I suggest we go with this now, and work on a full solution for revamping the errors on both levels in a different effort.

Reply:
Good point, but sad. let's leave it as it is.